pkg/controller/main-controller.go

```diff
@@ -1118,44 +1118,47 @@ func (c *Controller) syncHandler(key string) error {
 
 		us, err := adminClnt.ServerUpdate(ctx, updateURL)
 		if err != nil {
-			_ = c.removeArtifacts()
-
-			err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", tenantName, err)
-			if _, terr := c.updateTenantStatus(ctx, tenant, err.Error(), totalAvailableReplicas); terr != nil {
-				return terr
-			}
+			if err.(madmin.ErrorResponse).Code != "MethodNotAllowed" {
+				_ = c.removeArtifacts()
 
-			// Update failed, nothing needs to be changed in the container
-			return err
-		}
+				err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", tenantName, err)
+				if _, terr := c.updateTenantStatus(ctx, tenant, err.Error(), totalAvailableReplicas); terr != nil {
+					return terr
+				}
 
-		if us.CurrentVersion != us.UpdatedVersion {
-			// In case the upgrade is from an older version to RELEASE.2021-07-27T02-40-15Z (which introduced
-			// MinIO server integrated with Console), we need to delete the old console deployment and service.
-			// We do this only when MinIO server is successfully updated.
-			unifiedConsoleReleaseTime, _ := miniov2.ReleaseTagToReleaseTime("RELEASE.2021-07-27T02-40-15Z")
-			newVer, err := miniov2.ReleaseTagToReleaseTime(us.UpdatedVersion)
-			if err != nil {
-				klog.Errorf("Unsupported release tag on new image, server updated but might leave dangling console deployment %v", err)
+				// Update failed, nothing needs to be changed in the container
 				return err
 			}
-			consoleDeployment, err := c.deploymentLister.Deployments(tenant.Namespace).Get(tenant.ConsoleDeploymentName())
-			if unifiedConsoleReleaseTime.Before(newVer) && consoleDeployment != nil && err == nil {
-				if err := c.deleteOldConsoleDeployment(ctx, tenant, consoleDeployment.Name); err != nil {
+		}
+		if err == nil {
```

Comment:
This if condition not needed as we are returning from `if err != nil` part.

Reply:
No, if err is reject by minio-server which is started with 'MINIO_UPDATE=off', we ignore it.